MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs

```diff
@@ -287,24 +360,21 @@ private async Task<IList<Asset>> LoadAssetsFromGLTF(AssetSource source, Guid con
 		internal void OnAssetUpdate(AssetUpdate payload, Action onCompleteCallback)
 		{
 			var def = payload.Asset;
-			_app.AssetCache.OnCached(def.Id, asset =>
+			_app.AssetManager.OnSet(def.Id, asset =>
 			{
 				if (!_owner) return;
 
-				var mat = asset as UnityEngine.Material;
-				var tex = asset as UnityEngine.Texture;
-				if (def.Material != null)
+				if (def.Material != null && asset.Asset != null && asset.Asset is UnityEngine.Material mat)
 				{
+					// make sure material reference is write-safe
+					mat = _app.AssetManager.GetById(def.Id, writeSafe: true).Value.Asset as UnityEngine.Material;
```

Comment:
>GetById(def.Id, writeSafe: true).Value.Asset [](start = 29, length = 44)

Is this safe to dereference the value of the nullable when the GetById method can return null here and below?

Reply:
It is here, because this is inside the OnSet callback, so we are guaranteed that the asset already exists.